### ISSUE TITLE
Is it a way to know IP address of client ?

### ISSUE THREAD

**Issue by evgeshkastrong** [eclipse/mosquitto#2319]
it it possible in mqtt5 to know ip address of client and what technics to use for c++ ? 
1) when publisher connects to broker and want to know own external ip address...
2) when subscriber receive message from publisher and want to know his external ip address...

**Comment by Daedaluz** [comment]
I think you could use the plugin framework to add some properties onto messages going through the broker.
like, whenever someone pushes a message, add the ip of source to message.

if client want to know it's own ip, you could add a $control topic or some query topic to which the client can ask the server about the ip.

There is nothing built-in into mqttv5 protocol.

**Comment by evgeshkastrong** [comment]
thanks for your answer.   this is real useful feature which is absent by default - it would be great maybe in future releases to add this one.

**Comment by Daedaluz** [comment]
I don't think this will be added to any default setting because of the fact that it's not standard.
If required, it will probably have to be done through a plugin.

In what way would this be really useful for you?

**Comment by ralight** [comment]
@evgeshkastrong I don't see helping a client figure out its own network configuration as being related to MQTT at all really. As @Daedaluz says you could manage this with a plugin, but it's not something that will become part of the broker.
